metashare/api/sf_org_changes.py

```diff
@@ -0,0 +1,109 @@
+    )
+    repo_name, repo_owner = extract_owner_and_repo(repo_url)
+    repo_branch = "master"
+    cci = BaseCumulusCI(
+        repo_info={
+            "root": project_path,
+            "url": repo_url,
+            "name": repo_name,
+            "owner": repo_owner,
+            "commit": repo_branch,
+        }
+    )
+    package_xml_path = os.path.join(project_path, "src", "package.xml")
+    build_package_xml(scratch_org, package_xml_path, desired_changes)
+    task_config = TaskConfig(
+        {"options": {"path": project_path, "package_xml": package_xml_path}}
+    )
+    task = RetrieveUnpackaged(cci.project_config, task_config, org_config)
+    task()
+
+
+def commit_changes_to_github(
+    *, user, scratch_org, repo_url, branch, desired_changes, commit_message
+):
+    with local_github_checkout(user, repo_url) as project_path:
+        # This won't return anything in-memory, but rather it will emit files which we
+        # then copy into a source checkout, and then commit and push all that.
+        run_retrieve_task(user, scratch_org, project_path, desired_changes)
+        repo = get_repo_info(user, repo_url)
+        author = {"name": user.username, "email": user.email}
+        CommitDir(repo, author=author)(
+            project_path, branch, repo_dir="", commit_message=commit_message
+        )
+
+
+def get_salesforce_connection(*, config, login_url, base_url=""):
+    org_name = "dev"
+    org_config = refresh_access_token(
+        config=config, org_name=org_name, login_url=login_url
+    )
+
+    conn = simple_salesforce.Salesforce(
+        instance_url=org_config.instance_url,
+        session_id=org_config.access_token,
+        version=BaseGlobalConfig().project__package__api_version,
+    )
+    conn.headers.setdefault(
+        "Sforce-Call-Options", "client={}".format(settings.SF_CLIENT_ID)
+    )
+    conn.base_url += base_url
+
+    return conn
+
+
+def get_latest_revision_numbers(scratch_org):
+    conn = get_salesforce_connection(
+        config=scratch_org.config, login_url=scratch_org.login_url, base_url="tooling/"
+    )
+
+    # Store the results here on the org, and if any of these are > number than earlier
+    # version, there are changes.
+    # We need to run this right after the setup flow and store that as initial state.
+    records = conn.query_all(
+        "SELECT MemberName, MemberType, RevisionNum FROM SourceMember "
+        "WHERE IsNameObsolete=false"
+    ).get("records", [])
+
+    record_dict = defaultdict(lambda: defaultdict(dict))
+    for record in records:
+        record_dict[record["MemberType"]][record["MemberName"]] = record["RevisionNum"]
+
+    return record_dict
+
+
+def compare_revisions(old_revision, new_revision):
+    ret = defaultdict(list)
+    for mt in new_revision.keys():
+        for mn in new_revision[mt].keys():
+            if new_revision[mt][mn] > old_revision.get(mt, {}).get(mn, -1):
+                ret[mt].append(mn)
+    return ret
```

Comment:
@wlonk @davisagli Is it expected for this to return uncaptured changes for a newly-created Dev org, or is that a bug somewhere? When I create a brand new Dev org from https://github.com/oddbird/MetaShare-Test/ I get 16 uncaptured changes right away:

```json
{
  "latest_revision_numbers": {
    "Layout": {
      "SampleObject__c-SampleObject Layout": 19,
      "Child_Sample_Object__c-Child Sample Object Layout": 18
    },
    "Profile": {
      "Admin": 40
    },
    "WebLink": {
      "SampleObject__c.Just_Testing": 36,
      "Campaign.ViewCampaignInfluenceReport": 35
    },
    "ApexPage": {
      "SamplePage": 13
    },
    "ListView": {
      "SampleObject__c.All": 39,
      "Account.Household_Accounts": 37,
      "Opportunity.Default_Opportunity_Pipeline": 38
    },
    "ApexClass": {
      "SampleClass": 7,
      "SamplePage_CTRL": 9,
      "SampleClass_TEST": 8,
      "SamplePage_CTRL_TEST": 10
    },
    "CustomTab": {
      "SampleObject__c": 21
    },
    "RecordType": {
      "Account.HH_Account": 30,
      "Account.Organization": 31,
      "Opportunity.NPSP_Default": 32
    },
    "ApexTrigger": {
      "SampleTrigger": 11
    },
    "CustomField": {
      "SampleObject__c.Sample_Field__c": 26,
      "Child_Sample_Object__c.Parent__c": 25
    },
    "QuickAction": {
      "SampleObject__c.New_Child": 20
    },
    "CustomObject": {
      "SampleObject__c": 28,
      "Child_Sample_Object__c": 27
    },
    "ApexComponent": {
      "SampleComponent": 12
    },
    "CompactLayout": {
      "Account.NPSP_Household_Account": 33,
      "Account.NPSP_Organization_Account": 34
    },
    "HomePageLayout": {
      "SampleHomePage": 24
    },
    "BusinessProcess": {
      "Opportunity.NPSP_Default": 29
    },
    "CustomApplication": {
      "CumulusCITest": 22
    },
    "HomePageComponent": {
      "SampleComponent": 23
    }
  },
  "unsaved_changes": {
    "Profile": [
      "Admin"
    ],
    "WebLink": [
      "Campaign.ViewCampaignInfluenceReport",
      "SampleObject__c.Just_Testing"
    ],
    "ListView": [
      "Opportunity.Default_Opportunity_Pipeline",
      "SampleObject__c.All",
      "Account.Household_Accounts"
    ],
    "RecordType": [
      "Account.HH_Account",
      "Account.Organization",
      "Opportunity.NPSP_Default"
    ],
    "CustomField": [
      "Child_Sample_Object__c.Parent__c",
      "SampleObject__c.Sample_Field__c"
    ],
    "CustomObject": [
      "Child_Sample_Object__c",
      "SampleObject__c"
    ],
    "CompactLayout": [
      "Account.NPSP_Household_Account",
      "Account.NPSP_Organization_Account"
    ],
    "BusinessProcess": [
      "Opportunity.NPSP_Default"
    ]
  }
}
```

Reply:
I believe those shouldn't show up, so I've made it now initialize latest-changes after running the initial flow.